src/tool-myank.cpp

```diff
@@ -295,6 +306,46 @@ void Tool_myank::processFile(HumdrumFile& infile) {
 }
 
 
+vector<int> Tool_myank::analyzeBarNumbers(HumdrumFile& infile) {
+	vector<int> m_barnum;
+	m_barnum.resize(infile.getLineCount());
+	int current = -1;
+	HumRegex hre;
+	for (int i=0; i<infile.getLineCount(); i++) {
+		if (!infile[i].isBarline()) {
+			m_barnum.at(i) = current;
+			continue;
+		}
+		if (hre.search(infile[i].token(0), "=(\\d+)")) {
+			current = hre.getMatchInt(1);
+		}
+		m_barnum.at(i) = current;
+	}
+	return m_barnum;
+}
+
+int Tool_myank::getBarNumberForLine(int line) {
+	return BarNumbers[line-1];
+}
+
+int Tool_myank::getStartLine() {
+	regex re("^(\\d+)\\-(\\d+)$");
+    std::smatch match;
+	if (regex_search(linesQ, match, re) && match.size() > 1) {
+		return stoi(match.str(1));
+	}
+	return -1;
+}
+
+int Tool_myank::getEndLine() {
```

Comment:
Using std::regex is OK, but I typically use HumRegex for regular expressions in humlib:

```cpp
int Tool_myank::getEndLine() {
     HumRegex hre;
     if (hre.search(linesQ, "^(\\d+)\\-(\\d+)$")) {
         return hre.getMatchInt(2);
     } else {
          return -1;
     }
}
```

It also looks a bit more cleaner than using `std::regex`.